src/Maths/Silk.NET.Maths/Scalar.Log.cs

```diff
@@ -0,0 +1,308 @@
+﻿// This file is part of Silk.NET.
+// 
+// You may modify and distribute Silk.NET under the terms
+// of the MIT license. See the LICENSE file for details.
+
+/*
+ * Copyright (C) 2008-2020 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+using System;
+using System.Runtime.CompilerServices;
+#if SSE
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+#endif
+using static Silk.NET.Maths.Helper;
+
+namespace Silk.NET.Maths
+{
+    public static partial class Scalar
+    {
+        private static uint[] InitLog()
+        {
+#if POH
+            var logTblLookup = GC.AllocateUninitializedArray<uint>(771, true);
+#else
+            var logTblLookup = new uint[771];
+#endif
+            int i = 0;
+            logTblLookup[i++] = 0x40000000; logTblLookup[i++] =  0x0; logTblLookup[i++] =  0x0; logTblLookup[i++] =  0x3fff00ff; logTblLookup[i++] =  0x3b7f8000; logTblLookup[i++] =  0x32aa2b11; logTblLookup[i++] =  0x3ffe03f8; logTblLookup[i++] =  0x3bff0000; logTblLookup[i++] =  0x3429ac42; logTblLookup[i++] = 
```

Comment:
WOWZA. Can we reformat this file please?

Not urgent.

Reply:
I mean yes, but also I don't want to spend that many lines on that, I mean no one is interested in more then "it's a lookup table"

Reply:
Just tried formatting, I'd rather keep this. Rider/R# don't complain cause it's too much, and like this it at least doesn't completely dominate the file.